pkg/stockings/fingopack.go

```diff
@@ -0,0 +1,159 @@
+package stockings
+
+import (
+	"time"
+
+	"github.com/piquette/finance-go"
+	"github.com/piquette/finance-go/chart"
+	"github.com/piquette/finance-go/datetime"
+	"github.com/piquette/finance-go/quote"
+
+	"github.com/shopspring/decimal"
+)
+
+// FingoPack will be used to determine which interface we are using
+type FingoPack struct {
+}
+
+// fingoHistoricalStocks is a list of struct ChartBar which is determined by finance-go
+type piqHistoricalStocks []finance.ChartBar
+
+func (piq FingoPack) GetCurrentPrice(ticker string) (*Stock, error) {
+
+	// simple call to get the quote of a stock in the finance.quote struct
+	quote, err := quote.Get(ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	// need to convert our types to match our universal Stock struct - should this be broken out into a seperate function?
+	// some of the types are already correct, but listed them all out for clarity
+	// finance.quote has a HUGE variety of info in it that we can pull
+	symbol := quote.Symbol
+	name := quote.ShortName
+	currentPrice := decimal.NewFromFloat(quote.RegularMarketPrice)
+	priceChange := float32(quote.RegularMarketChange)
+	percentChange := float32(quote.RegularMarketChangePercent)
+
+	return &Stock{symbol, name, currentPrice, priceChange, percentChange}, nil
+
+}
+
+// GetHistoricalPrice is hacky because Fingo only has a function that returns quotes in a time span
+// This function takes in the desired date as the end date, then sets the start date as one month prior
+func (piq FingoPack) GetHistoricalPrice(ticker string, date time.Time) (*HistoricalStock, error) {
```

Comment:
also, do you realize that `GetHistoricalPrice` is essentially doing what `GetHistoricalRange` is doing? :) can we simplify the two functions to share like 90% of the same code?

Reply:
created a helper function that takes in the ticker and dates and returns the iterator

Reply:
lol do you still not see for `GetHistoricalPrice` you can just literally call `GetHistoricalRange` where you feed in `ticker`, `date` (as endDate), and `date.AddDate(0, -1, 0)` (as startDate) ? like everything below this line can get cleaned up, and your `GetHistoricalPrice` is like two lines